src/main/webapp/js/AddNewAnnotation.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Add a new annotation to a line
+ */
+class AddNewAnnotation {
+  /**
```

Comment:
nit:I think AddAnnotation is a better name, since it is an Add "New" is implied. Can leave it as is for now